packages/base/src/dialogs/symbology/vector_layer/types/Heatmap.tsx

```diff
@@ -63,11 +63,24 @@ const Heatmap: React.FC<ISymbologyDialogProps> = ({
       return;
     }
 
-    const colorMap = colormap({
-      colormap: selectedRampRef.current,
-      nshades: 9,
-      format: 'hex',
-    });
+    const cmoRamp = (cmocean as any)[selectedRampRef.current];
+    let colorMap: string[];
+
+    if (cmoRamp) {
+      const fullRamp = colormap({
+        colormap: selectedRampRef.current,
+        nshades: cmoRamp.length,
+        format: 'hex',
+      });
+      const step = Math.floor(fullRamp.length / 9);
+      colorMap = Array.from({ length: 9 }, (_, i) => fullRamp[i * step]);
+    } else {
+      colorMap = colormap({
+        colormap: selectedRampRef.current,
+        nshades: 9,
+        format: 'hex',
+      });
+    }
```

Comment:
Do we need this special case for cmocean colormaps? My goal with the change to updating the `colormap` `colorScale` is so we shouldn't have any special cases.